capellambse/diagram/_icons.py

```diff
@@ -0,0 +1,1240 @@
+    )
+    letter.add(path.Path(d=d, stroke_width=0.1))
+    symb.add(letter)
+    return symb
+
+
+@_factory()
+def _physical_node_human_actor_symbol() -> container.Symbol:
+    return standalone_stick_figure_symbol("PhysicalNodeHumanActorSymbol")
+
+
+def _make_port_box(id_: str) -> container.Symbol:
+    port = container.Symbol(id=id_, viewBox="0 0 10 10")
+    port.add(path.Path(d="M0 0h10v10H0z"))
+    return port
+
+
+def _make_edge_symbol(
+    id_: str, grad_colors: tuple[str, str], middle_color: str
+) -> container.Symbol:
+    symb = container.Symbol(id=id_, viewBox="0 0 40 30")
+    grad_id = id_ + "-gradient"
+    symb.add(
+        _make_lgradient(
+            grad_id,
+            stop_colors=grad_colors,
+            start=(0, 0),
+            end=(1, 1),
+        )
+    )
+    symb.add(
+        _make_lgradient(
+            f"{grad_id}reverse",
+            stop_colors=grad_colors[::-1],
+            start=(0, 0),
+            end=(1, 1),
+        )
+    )
+
+    grp = container.Group(stroke="#000", stroke_width=2)
+    grp.add(
+        path.Path(
+            d="M 36.190065,5.0377724 V 24.962228 H 26.17482 V 5.0377724 Z",
+            fill=f"url(#{grad_id}reverse)",
+        )
+    )
+    d = (
+        "m 14.372107,10 h 12.622435 c 0.926189,0.585267"
+        " 1.836022,1.274509 2.268178,5 -0.208657,2.812473"
+        " -0.954601,4.503809 -2.273297,5 H 14.296948"
+    )
+    grp.add(path.Path(d=d, fill=middle_color))
+    d = (
+        "M 3.9464908,5.0048246 V 24.995175 H 10.87518 C"
+        " 12.433713,24.159139 15.158267,20.291241 15.313795,15"
+        " 15.498614,11.583142 14.059659,6.6240913 10.87518,5.0048246 c"
+        " -2.2179509,0 -4.5908341,0 -6.9286892,0 z"
+    )
+    grp.add(path.Path(d=d, fill=f"url(#{grad_id})"))
+    symb.add(grp)
+    return symb
+
+
+def _make_lgradient(
+    id_,
+    /,
+    *,
+    start=(0, 0),
+    end=(0, 1),
+    stop_colors=("#fff", "#000"),
+    stop_opacity=(1, 1),
+) -> gradients.LinearGradient:
+    if len(start) != 2 or len(end) != 2:
+        raise ValueError(
+            "Exactly two values each for start and end are needed"
+        )
+
+    offsets = [i / (len(stop_colors) - 1) for i in range(len(stop_colors))]
+    grad = gradients.LinearGradient(id_=id_, start=start, end=end)
+    for offset, stop_col, stop_op in zip(offsets, stop_colors, stop_opacity):
+        grad.add_stop_color(offset=offset, color=stop_col, opacity=stop_op)
+    return grad
+
+
+def _make_rgradient(
+    id_,
+    /,
+    *,
+    center=(0, 0),
+    r=1,
+    focal=(0, 0),
+    matrix=None,
+    stop_colors=("#fff", "#000"),
+    offsets=("0", "1"),
+) -> gradients.RadialGradient:
+    grad = gradients.RadialGradient(
+        id_=id_,
+        center=center,
+        r=r,
+        focal=focal,
+        gradientUnits="userSpaceOnUse",
+        gradientTransform=(
+            f"matrix({' '.join(map(str, matrix))})"
+            if matrix is not None
+            else None
+        ),
+    )
```

Comment:
```suggestion
    params = {
        "id_": id_,
        "center": center,
        "r": r,
        "focal": focal,
        "gradientUnits": "userSpaceOnUse",
    }
    if matrix is not None:
        params["gradientTransform"] = f"matrix({' '.join(map(str, matrix))})"

    grad = gradients.RadialGradient(**params)
```

Somehow this fails if matrix is `None`.

Reply:
I might have found a better one, can you please retest against https://github.com/DSD-DBS/py-capellambse/pull/424/commits/d093c529742e4edaab4ab02cf59a9b058b9c230f?

Reply:
We tested it together, and it works fine.